src/plugins/login/azureLoginPlugin.ts

```diff
@@ -24,12 +28,13 @@ export class AzureLoginPlugin extends AzureBasePlugin<AzureLoginOptions> {
     this.log("Logging into Azure");
 
     try {
```

Comment:
Since you aren't using the token cache outside of this function or class I would recommend moving the token cache and the call to `addSubs(...)` into the Login service itself to reduce the complexity here.